R/01-qc.R

```diff
@@ -1,119 +1,50 @@
-#' @param plasmid_cutoff default it NULL; `qc_plasmid_histogram` will calculate and set a default one unless a value is specified here 
+#' @param plasmid_cutoff default it NULL; `qc_plasmid_histogram` will calculate and set a default one unless a value is specified here
 #' @param use_combined default it TRUE; if TRUE, both zero count and low plasmid CPM filters are applied and if either is TRUE, a pgRNA construct will be filtered out. If FALSE, need to allow to specify which should be used
-#' @param params default is `list(cell_line = NULL)`; used to pass info when saving plots
 #' @param plots_dir default is `./qc_plots`; directory to save plots created with this function, if it doesn't exist already it will be created
 #' @param overwrite default is FALSE; whether to overwrite the QC Report file
-#' @param output_format default is NULL; the output format that the QC Rmd will be knitted to/produced
-#' @param output_file default is `QC_Report`; name of the output QC report file
-#' @param output_dir default is `./`; where the qc report will be stored
-#' @param wide_ar default is 0.75; aspect ratio
+#' @param output_file_path default is `QC_Report`; name of the output QC report file
+#' @param ...
 #' @export
 #' @importFrom tidyr pivot_longer
 #' @importFrom magrittr %>%
 #' @examples \dontrun{
 #'
 #' }
+run_qc <- function(gimap_dataset,
+                   plots_dir = "./qc_plots",
+                   overwrite = FALSE,
+                   output_format = NULL,
+                   output_file = "QC_Report",
+                   output_dir = "./",
+                   plot_suffix = "png") {
 
-
-run_qc <- function(gimap_dataset, plasmid_cutoff = NULL, use_combined = TRUE,
-                   params = list(cell_line = NULL), plots_dir = "./qc_plots", overwrite = FALSE, output_format = NULL, output_file = "QC_Report", output_dir = "./",
-                   wide_ar = 0.75) {
-  
-  store_results <- function(result_obj, lvoi, list_of_qc_things, plots_dir, params){
-    pdf_base <- file.path(plots_dir, "plots", "pdf")
-    png_base <- file.path(plots_dir, "plots", "png")
-    list_of_qc_things[[lvoi]] <- result_obj
-    save_plot(result_obj, params=params, out_dir = plots_dir)
-    list_of_qc_things[[paste0(eval(lvoi), "_file_loc")]][["pdf"]] <- file.path(pdf_base, paste0(params$cell_line, "_", deparse(substitute(result_obj)), ".pdf"))
-    list_of_qc_things[[paste0(eval(lvoi), "_file_loc")]][["png"]] <- file.path(png_base, paste0(params$cell_line, "_", deparse(substitute(result_obj)), ".png"))
-    return(list_of_qc_things)
-  }
-  
-  if (!dir.exists(plots_dir)){
+  if (!dir.exists(plots_dir)) {
     dir.create(plots_dir, showWarnings = TRUE)
   }
-  
+
   if (!("gimap_dataset" %in% class(gimap_dataset))) stop("This function only works with gimap_dataset objects which can be made with the setup_data() function.")
-  
-  # for Rmd report
-  list_of_qc_things <- list(counts_cdf = NULL,
```

Comment:
I think this totally made sense as a first attempt! It makes sense that you may have been following the structure in the setup_data.R code where I had written a structured list where data was set up. 

Is there a reason you thought we needed to follow that same setup here instead of going to the Rmd directly? 

Again, thanks so much for getting this first setup ready. Your digging in on the biology was really superb.

Reply:
I thought that the Rmarkdown/report was a secondary goal and would be an option that people could choose whether or not to render. I didn't plan for the overall `list_of_qc_thing` to get returned and passed on to the next function of (finding log fold change), rather only returning the filter that I saved within the gimap dataset for the next function. This list instead served as a way to save ggplot objects and strings (of where plots were saved) or values of certain parameters for the report and pass it to a separate generate report function. That way, all the info was stored as it was used/produced in real time of running QC, and if someone chose to make the report, it was just filling in info rather than rerunning things.

Reply:
If you don't want the report to be optional, it makes sense to do everything in the rmarkdown as long as the end filter is available for the next step.